apps/worker/src/monitor/tx.status.monitor.service.ts

```diff
@@ -85,42 +79,47 @@ export class TxStatusMonitoringService extends WorkerHost implements OnApplicati
-    const txReceiptPromises: Promise<BlockHash | undefined>[] = blockList.map(async (blockNumber) => {
-      const blockHash = await this.blockchainService.getBlockHash(blockNumber);
-      const block = await this.blockchainService.getBlock(blockHash);
-      const txInfo = block.block.extrinsics.find((extrinsic) => extrinsic.hash.toString() === txHash.toString());
-      this.logger.debug(`Extrinsics: ${block.block.extrinsics[0]}`);
+  private async handleMessagesFailure(jobId: string, moduleError: RegistryError): Promise<void> {
+    this.logger.debug(`Handling extrinsic failure for job ${jobId} and error ${JSON.stringify(moduleError)}`);
+    const txJob = (await this.txReceiptQueue.getJob(jobId)) as Job<ITxMonitorJob, any, string>;
+    if (!txJob) {
+      this.logger.error(`Job ${jobId} not found in tx receipt queue`);
+    }
+    const blockDelay = 1 * SECONDS_PER_BLOCK * MILLISECONDS_PER_SECOND;
 
-      if (txInfo !== undefined) {
-        this.logger.verbose(`Found tx ${txHash} in block ${blockNumber}`);
-        const at = await this.blockchainService.api.at(blockHash.toHex());
-        const events = await at.query.system.events();
-        events.subscribe((records) => {
-          records.forEach(async (record) => {
-            const { event } = record;
-            const eventName = event.section;
-            const { method } = event;
-            const { data } = event;
-            this.logger.debug(`Received event: ${eventName} ${method} ${data}`);
-            if (eventName.search('capacity') !== -1 && method.search('Withdrawn') !== -1) {
-              const capacityWithDrawn = BigInt(data[1].toString());
-              this.logger.debug(`Capacity withdrawn: ${capacityWithDrawn}`);
-              this.setEpochCapacity(epoch, capacityWithDrawn);
-            }
-          });
-        });
-        return blockHash;
+    try {
+      switch (moduleError.name) {
+        case 'TooManyMessagesInBlock':
+          // Re-try the job in the publish queue with a block delay
+          await this.publishQueue.removeRepeatableByKey(txJob.data.referencePublishJob.id);
+          await this.publishQueue.add(txJob.data.referencePublishJob.id, txJob.data.referencePublishJob, { delay: blockDelay });
+          break;
+        case 'InvalidMessageSourceAccount':
+          // PROVIDER_ID is not set correctly
+          // Pause publishing entirely
+          this.publishQueue.pause();
+          break;
+        case 'InvalidSchemaId':
+          // Fail the job since this should never happen, or is a protocol error
+          // pause publishing entirely
+          this.publishQueue.pause();
+          break;
+        case 'UnAuthorizedDelegate':
+          // Re-try the job in the publish queue with a block delay, could be a signing error
+          await this.publishQueue.removeRepeatableByKey(txJob.data.referencePublishJob.id);
```

Comment:
Do we need to fail the job or repeat? If there are no authorization is there a scenario that they would get authorized in the delay interval?

Reply:
I think we we will re-try the publishJob if it fails again n times as attempts allowed, it should be fine, given if this is related to signature expired for some reason then it will try again, and if its failing due to bad Provider keys, then it will fail for everything, 

Assuming after 3/n attempts it wont try